backend/handler/share/comment.go

```diff
@@ -84,11 +84,9 @@ func (h *ShareCommentHandler) CreateComment(c echo.Context) error {
 	if !appInfo.Settings.WebAppCommentSettings.IsEnable {
 		return h.NewResponseWithError(c, "please check comment is open", nil)
 	}
-	// validate captcha token if captcha is enable for comment
-	if appInfo.Settings.CaptchaSettings.ChatStatus == consts.CaptchaStatusEnable {
-		if !h.Captcha.ValidateToken(ctx, req.CaptchaToken) {
-			return h.NewResponseWithError(c, "failed to validate captcha token", nil)
-		}
+	// validate captcha token
+	if !h.Captcha.ValidateToken(ctx, req.CaptchaToken) {
```

Comment:
同chat.go中的问题，在移除基于应用设置的验证码验证逻辑后，所有评论请求都会无条件验证验证码。这可能会导致在不需要验证码的场景下出现不必要的验证步骤，建议恢复基于设置的条件验证逻辑，或提供其他机制来控制验证码的使用。

```suggestion
// validate captcha token if captcha is enabled
if shouldValidateCaptcha(appInfo) && !h.Captcha.ValidateToken(ctx, req.CaptchaToken) {
    return h.NewResponseWithError(c, "failed to validate captcha token", nil)
}